### PR TITLE
Change main method under dubbo-remoting-api to standard test

### PR DESCRIPTION
**Pull request by xiaomoran**
## What is the purpose of the change
Repair main method under dubbo-remoting-api due to dependency and change it to standard test method

image

### CODE REVIEW

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java [hunk 1]
```diff
@@ -14,93 +14,116 @@
...
+    private ExchangeServer exchangeServer;
+
+    private ConcurrentHashMap<String, ExchangeChannel> clients = new ConcurrentHashMap<>();
 
-    private static void startServer(int port) throws Exception {
+    private int port;
+
+    @BeforeEach
+    public void startServer() throws RemotingException {
+        port = (int) (1000 * Math.random() + 10000);
```
Comment (htynkn) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: if you are looking for a random port (which is available), check `NetUtils.getAvailablePort()`

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java [hunk 2]
```diff
@@ -14,93 +14,116 @@
...
-                    try {
-                        test(port);
-                    } catch (Exception e) {
-                        e.printStackTrace();
-                    }
+            exec.execute(() -> {
+                try {
+                    clientExchangeInfo(port);
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (htynkn) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: for any exception, we should consider this unit test fail.
You can throw this exception or invoke `fail` method in junit library

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java [hunk 3]
```diff
@@ -14,93 +14,116 @@
...
-            public Object reply(ExchangeChannel channel, Object msg) {
-                for (int i = 0; i < 10000; i++)
-                    System.currentTimeMillis();
-                System.out.println("handle:" + msg + ";thread:" + Thread.currentThread().getName());
-                return new StringMessage("hello world");
+        dispatcher.addReplier(Object.class, (channel, msg) -> {
+            for (int i = 0; i < 10000; i++) {
+                System.currentTimeMillis();
             }
+            System.out.println("handle:" + msg + ";thread:" + Thread.currentThread().getName());
```
Comment (htynkn) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: This sout is not required for unit test. maybe remove it

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java [hunk 4]
```diff
@@ -14,93 +14,116 @@
...
-        System.out.println("send finished.");
+        clients.put(Thread.currentThread().getName(), client);
     }
 
-    static void mutliThreadTest(int tc, final int port) throws Exception {
-        Executor exec = Executors.newFixedThreadPool(tc);
+
+    @Test
+    public void testMutliThread() throws Exception {
+        int tc = 10;
```
Comment (htynkn) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: `Mutli` --> `Multi` ?

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java (hunk 1 above)
Comment (xiaomoran) [reply dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: Thank you for your correction

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java (hunk 4 above)
Comment (xiaomoran) [reply dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/ReplierDispatcherTest.java]: Thank you for your suggestion. I've already revised it.

**Pull request closed without merging**